sklearn/manifold/t_sne.py

```diff
@@ -423,18 +423,30 @@ def trustworthiness(X, X_embedded, n_neighbors=5, precomputed=False):
         dist_X = X
     else:
         dist_X = pairwise_distances(X, squared=True)
-    ind_X = np.argsort(dist_X, axis=1)
+    np.fill_diagonal(dist_X, np.inf)  # excludes dist(x_i, x_i)
+    ind_X = np.argsort(dist_X, axis=1)[:, :-1]
+    # ind_X[i] now is the ordered list of k-neighbors of i in the input space
     ind_X_embedded = NearestNeighbors(n_neighbors).fit(X_embedded).kneighbors(
         return_distance=False)
 
+    # We build an inverted index returning the rank of sample j w.r.t sample i
```

Comment:
I find this explanation rather obscure, and the example not really helping.
I suggest something explaining the relations between `dist_X`, `ind_X` and `inverted_index`.
For example:
```py
    # We build an inverted index of neighbors in the input space: For sample i,
    # we define `ind_X[i]` as the index of sorted distances to other samples:
    #         ind_X[i] = np.argsort(dist_X[i])[1:]
    # i.e.    dist_X[i][ind_X[i]] = np.sort(dist_X[i])[1:]
    # and `inverted_index[i]` as the inverted index of sorted distances:
    #         inverted_index[i][ind_X[i]] = np.arange(n_sample)[1:]
    # i.e.    ind_X[i] = np.argsort(inverted_index[i])[1:]
```

Reply:
I agree it was obscure, 
I changed it with your proposal which is clearer indeed